examples/todomvc/components/TodoItem.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import TodoTextInput from './TodoTextInput';
+
+
+export default class TodoItem extends Component {
+  static propTypes = {
+    todo: PropTypes.object.isRequired,
+    editTodo: PropTypes.func.isRequired,
+    deleteTodo: PropTypes.func.isRequired,
+    markTodo: PropTypes.func.isRequired
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      edit: false
+    };
+  }
+
+  onDoubleClick() {
+    this.setState({ edit: true });
+  }
+
+  onSave(id, text) {
+    if (text.length === 0) {
+      this.props.deleteTodo(id);
+    } else {
+      this.props.editTodo(id, text);
+    }
+    this.setState({ edit: false });
+  }
+
+  render() {
+    const {todo, markTodo, deleteTodo, editTodo} = this.props;
+
+    let view;
+
+    if (this.state.edit) {
+      view = (<TodoTextInput
```

Comment:
Style nit pick: please put newline after the `(` in JSX:

```js
  view = (
    <TodoTextInput attr=...
                   anotherAttr=... />
  );
```

This way they're all intended the same.

Also please change `view` to `element` or `input`.. Don't want to sound MVCish there.